pkg/aws/ocpgwdeployer.go

```diff
@@ -198,18 +220,29 @@ type machineSetConfig struct {
 	Region        string
 	SecurityGroup string
 	PublicSubnet  string
+	NodeSGSuffix  string
 }
 
 func (d *ocpGatewayDeployer) findAMIID(vpcID string) (string, error) {
-	result, err := d.aws.client.DescribeInstances(context.TODO(), &ec2.DescribeInstancesInput{
-		Filters: []types.Filter{
-			ec2Filter("vpc-id", vpcID),
-			d.aws.filterByName("{infraID}-worker*"),
-			d.aws.filterByCurrentCluster(),
-		},
-	})
-	if err != nil {
-		return "", errors.Wrap(err, "error describing AWS instances")
+	ownedFilters := d.aws.filterByCurrentCluster()
+	var err error
+	var result *ec2.DescribeInstancesOutput
+
+	for i := range ownedFilters {
+		result, err = d.aws.client.DescribeInstances(context.TODO(), &ec2.DescribeInstancesInput{
+			Filters: []types.Filter{
+				ec2Filter("vpc-id", vpcID),
+				d.aws.filterByName("{infraID}-worker*"),
+				ownedFilters[i],
+			},
+		})
+		if err != nil {
```

Comment:
Do we want to return error if we fail to describe instance for single-owned filter or should we log error and continue with the for loop?

Reply:
it could be a real error should we ignore in that case?

Reply:
Well, it's corner case, but I guess if it's enough to find a valid instance for a single-owned filter, we can log the error and continue with the for loop.

I assume we should return error here only if we could not find a valid result for any ownedFilter.

Reply:
There is no logger in cloud-prepare now , we can continue and ignore.